screens/EventDetails/ExpandableText.tsx

```diff
@@ -0,0 +1,84 @@
+import { BodyText, Headline } from "@components/Text"
+import { ButtonStyles } from "@lib/ButtonStyle"
+import React, {useEffect, useRef, useState } from "react"
+import { Animated, Button, TextProps, View } from "react-native"
+
+export type ExpandableTextProps = {
+  props: TextProps
+  text: string
+  linesToDisplay: number
+}
+
+const TEXT_HEIGHT = 22
+
+const ExpandableText = ({
+  props,
+  text,
+  linesToDisplay
+}: ExpandableTextProps) => {
+  const [expander, setExpander] = useState(false)
+  const [expanded, setExpanded] = useState(false)
+  const [lineNumber, setLineNumber] = useState(linesToDisplay)
+  const [lineHeight, setLineHeight] = useState(TEXT_HEIGHT)
+  const startingHeight = lineHeight * linesToDisplay
+  const [fullHeight, setFullHeight] = useState(startingHeight)
+  const animatedHeight = useRef(new Animated.Value(startingHeight)).current
+
+  const toggleExpansion = () => {
+    setExpanded(!expanded)
+  }
+
+  useEffect(() => {
+    Animated.spring(animatedHeight, {
+      friction: 100,
+      toValue: expanded ? fullHeight : startingHeight,
+      useNativeDriver: false
+    }).start()
+  }, [expanded])
+
+  useEffect(() => {
+    if (lineNumber < linesToDisplay) {
+      animatedHeight.setValue(lineHeight * lineNumber)
```

Comment:
Thanks for encapsulating everything, helps a lot with organization and readability